tests/jobs/localisation/test_integration.py

```diff
@@ -0,0 +1,262 @@
+                },
+            },
+            {
+                "name": "CORR2",
+                "obs_group": {"add": "*", "remove": obs_group_add},
+                "param_group": {
+                    "add": "SNAKE_OIL_PARAM:*",
+                    "remove": "SNAKE_OIL_PARAM:OP1_PERSISTENCE",
+                },
+            },
+            {
+                "name": "CORR3",
+                "obs_group": {"add": obs_group_add},
+                "param_group": {
+                    "add": param_group_add,
+                },
+            },
+        ],
+    }
+    with open("local_config.yaml", "w") as fout:
+        yaml.dump(config, fout)
+    LocalisationConfigJob(ert).run("local_config.yaml")
+    assert ert.getLocalConfig().getMinistep("CORR1").name() == "CORR1"
+    assert (
+        ert.getLocalConfig().getObsdata("CORR1_obs_group").name() == "CORR1_obs_group"
+    )
+    result = {}
+    for index, ministep in enumerate(ert.getLocalConfig().getUpdatestep()):
+        result[ministep.name()] = {
+            "obs": [obs_node.key() for obs_node in ministep.getLocalObsData()],
+            "key": ministep.name() + "_param_group",
+        }
+    expected_result = {
+        "CORR1": {
+            "obs": [
+                "WOPR_OP1_108",
+                "WOPR_OP1_144",
+                "WOPR_OP1_36",
+                "WOPR_OP1_72",
+                "WOPR_OP1_9",
+                "WPR_DIFF_1",
+            ],
+            "key": "CORR1_param_group",
+        },
+        "CORR2": {
+            "obs": [
+                "WOPR_OP1_108",
+                "WOPR_OP1_144",
+                "WOPR_OP1_36",
+                "WOPR_OP1_72",
+                "WOPR_OP1_9",
+                "WPR_DIFF_1",
+            ],
+            "key": "CORR2_param_group",
+        },
+        "CORR3": {"obs": ["FOPR", "WOPR_OP1_190"], "key": "CORR3_param_group"},
+    }
+    assert result == expected_result
+
+
+def test_localisation_gen_param(
+    setup_poly_ert,
+):
+    with open("poly.ert", "a") as fout:
+        fout.write(
+            "GEN_PARAM PARAMS_A parameter_file_A INPUT_FORMAT:ASCII "
+            "OUTPUT_FORMAT:ASCII INIT_FILES:initial_param_file_A_%d"
+        )
+    nreal = 5
+    nparam = 10
+    for n in range(nreal):
+        filename = "initial_param_file_A_" + str(n)
+        with open(filename, "w") as fout:
+            for i in range(nparam):
+                fout.write(f"{i}\n")
+
+    res_config = ResConfig("poly.ert")
+    ert = EnKFMain(res_config)
+    config = {
+        "log_level": 2,
+        "correlations": [
+            {
+                "name": "CORR1",
+                "obs_group": {
+                    "add": "*",
+                },
+                "param_group": {
+                    "add": "*",
+                },
+            },
+        ],
+    }
+
+    with open("local_config.yaml", "w") as fout:
+        yaml.dump(config, fout)
+    LocalisationConfigJob(ert).run("local_config.yaml")
```

Comment:
Is it possible to assert something after this test?

Reply:
As long as the ert model is not run but only the localisation config job, the only I know about can be asserted is that the object created and updated for ministeps contain the correct parameter nodes, parameter names, observation names and which parameters are active or not. No check of the effect can be asserted unless the ert is run through an update step with ensemble smoother and we can compare the result with the expected result we know is correct. Any verification of correctness is difficult (visualisation of e.g fields can be one solution) except for very simple (and often uninterested cases), But it is of course possible to verify that an expected result is reproduced.  Maybe we can include such a test (compare with reference) later ?